torch/amp/grad_scaling.py

```diff
@@ -0,0 +1,382 @@
+            populated.
+
+    Arguments:
+        current_scale (float or torch.cuda.FloatTensor):  The value most recently used to scale gradients
+            (by e.g. a call to :func:`amp.scale_outputs`).
+        scale_growth_factor (float, optional, default=1.001):  A Python float to be used as the scale growth factor.
+            If ``scale_scheduler`` is supplied, this argument will be ignored.
+        scale_backoff_factor (float, optional, default=0.5):  A Python float to be used as the scale backoff factor.
+            If ``scale_scheduler`` is supplied, this argument will be ignored.
+        scale_scheduler (callable, optional):  Function to customize how the recommended scale is updated.
+            If supplied, ``scale_scheduler`` should take 2 arguments.  Both will be one-element
+            ``torch.cuda.FloatTensors``.  The first will contain the current scale value.  The second will contain > 0
+            if gradients currently contain inf or nan, and ``0.0`` otherwise.  The scheduler should use these arguments
+            to determine the next recommended scale, and return a one-element ``torch.cuda.FloatTensor`` containing the
+            next recommended scale.
+        skip_if_inf (bool, optional, default=True):  If False, unscale gradients but call ``step()`` blindly without
+            checking if the gradients contain inf or nan.  If you're manually using the same gradient scale every
+            iteration ("static gradient scaling", as opposed to dynamic gradient scaling) and you're fairly
+            certain no inf/nan gradients will be encountered, setting ``skip_if_inf=False`` can provide a minor
+            performance improvement, because for typical optimizers it avoids copying ``found_inf``'s value back to
+            the CPU.
+            Another case where it makes sense to supply ``skip_if_inf=False`` is batch replay, where you replay this
+            iteration until inf/nan-free gradients are created.  By the time you reach the
+            :func:`optimizer.unscale_and_step()` call, you're sure the gradients do not contain inf/nan, so it's safe
+            to elide the check.  For other (typical) use, the default ``skip_if_inf=True`` is strongly recommended.
+        found_inf (torch.cuda.FloatTensor, optional):  If you've already created a 1-element Tensor that contains > 0.0
+            if the optimizer's current gradients have an inf/nan and 0.0 if they don't (by, for example, an earlier
+            call to ``optimizer.check_inf()``) you may pass this as ``found_inf``.  By default, you don't need to pass
+            ``found_inf`` manually.  :func:`unscale_and_step` will inspect the gradients for you, create a
+            ``found_inf`` tensor, and return it.  Leaving ``found_inf=None`` and letting :func:`unscale_and_step`
+            create it for you is typical.
+        scaling_enabled (bool, optional):  If ``False``, :func:`unscale_and_step` simply calls ``self.step()`` and
```

Comment:
Would passing a scale of 1 be an acceptable surrogate for this?

Reply:
No, because 1.0 is still a valid loss scale value even when doing dynamic loss scaling.  The scaling logic is still obligated to pass over the params to check for infs/nans even if the scale is 1.0.

Reply:
What about None?